sklearn/utils/testing.py

```diff
@@ -859,7 +859,7 @@ def check_docstring_parameters(func, doc=None, ignore=None):
     return incorrect
 
 
-def assert_run_python_script(source_code, timeout=60):
+def assert_run_python_script(source_code, timeout=60, ignore_warnings=True):
```

Comment:
I modified a bit this helper to be able to check the output without the warnings when we are only looking for errors.

Reply:
Shouldn't it be False by default? I feel like it will break backward compatibility of e.g. `test_import_is_deprecated` (though I'm surprised the checks arent failing already)

Reply:
Also you only use it with False here